polyfills.js

```diff
@@ -132,7 +132,7 @@ function patch (fs) {
     }
 
     // This ensures `util.promisify` works as it does for native `fs.read`.
-    read.__proto__ = fs$read
+    if (Object.setPrototypeOf) Object.setPrototypeOf(read, fs$read)
```

Comment:
This breaks engines that have dunder proto but lack setPrototypeOf.

Reply:
The only purpose of setting `__proto__` is to re-expose the `Symbol(customPromisifyArgs)` property of `fs.read` in newer versions of node.js (8+ I think).  I'm not aware of any platform with `util.promisify` but no `Object.setPrototypeOf`.

Reply:
https://npmjs.com/util.promisify can be polyfilled on any platform, including older nodes, but I take your point that it's less of an issue here.

Reply:
@ljharb That would be Node.js 0.10 and below, afaik.

I checked https://npmjs.com/util.promisify and I don't think it does anything with `fs.read`, so it behavior won't be affected by this change.

`util.promisify` is already broken with `fs.read` on Node.js 0.10.x, so thins won't affect anything there.

There is no `Symbol(customPromisifyArgs)` in Node.js 0.10.

Reply:
I mentioned that in this part of the PR description:
> There is no reason to set `.read()` proto when `Object.setPrototypeOf` is not available, as no special handling of `util.promisify` exists on those versions (even if `util.promisify` is shipped separately somehow), so no `__proto__` fallback there.

Reply:
To test (save as `test.js`):
```js
if (!global.Promise) global.Promise = require('bluebird')
var promisify = require('util.promisify')
var fs = require('fs')
var method = promisify(fs.read)
var fd = fs.openSync('./test.js', 'r')
method(fd, Buffer(100), 0, 100, 0).then(function(a, b, c) {
  console.log([a, b, c])
})
```